common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/OutLinedTextInput.kt

```diff
@@ -220,6 +231,17 @@ class OutLinedTextInputView @JvmOverloads constructor(
         doOnNextLayout { isHintAnimationEnabled = hintAnimationEnabledSnapshot }
     }
 
+    private fun TypedArray.getInputType(): Type {
+        val inputTypeInt = getInt(styleable.OutLinedTextInputView_type, INPUT_TYPE_MULTI_LINE.value)
+        return Type.values().firstOrNull { it.value == inputTypeInt } ?: INPUT_TYPE_MULTI_LINE
```

Comment:
I was slightly confused by this but I think `Type.values()[inputTypeInt]` should be more readable and simpler.

Reply:
`Type.values()[inputTypeInt]` changes the behavior in an undesirable way. 
- That relies on the _index_ to do the matching, but isn't matching on the actual specified _value_. 
- The way we have the code now, both the `value` and `index` match up, so your code would run fine. but if we had to add attributes with values not in order, or skipped some, then `index` and `value` wouldn't match up.
- If somehow something doesn't match up, your suggestion throws an exception rather than nulling out and letting us `?:` for a safe default

Thinking more, it might make sense to not even specify a value explicitly and rely on ordering alone, but I'm going to stick with the existing, explicit code for now.